### DISCUSSION TITLE
[ERROR] Failed to execute goal org.yamcs:yamcs-maven-plugin:1.3.2:protoc (default) on project yamcs-api:

### DISCUSSION THREAD

**Discussion by JohnAdam-123** [yamcs/yamcs#955] [Q&A]
How do I solve this problem, I was installing yamcs in windows 11 and unfortunate the project failed to build. _protoc did not exit cleanly. Review output for more information. -> [Help 1]_ but also here is another output _yamcs_out: protoc-gen-yamcs: Plugin failed with status code 1_. 
![Uploading maven_plugin_error.png…]()

**Comment by fqqb** [comment]
I can't see your picture. There's no known problems with v1.3.2 on Windows.
Maybe run mvn with `-X` to capture more error output.

**Reply by JohnAdam-123** [reply]
Yes, I did that already. But I couldn't manage to solve it.

_[ERROR] Failed to execute goal org.yamcs:yamcs-maven-plugin:1.3.2:protoc (default) on project yamcs-api: protoc did not exit cleanly. Review output for more information. -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.yamcs:yamcs-maven-plugin:1.3.2:protoc (default) on project yamcs-api: protoc did not exit cleanly. Review output for more information._

In short that's what if generally outputs, though I can share the whole debug info if you want to see.

**Reply by JohnAdam-123** [reply]
Here is the few lines of the output... it goes all the way to all the file in protoc folder I think.

_[INFO] Compiling 50 proto file(s) to C:\Program Files\yamcs\yamcs-master\yamcs-api\target\generated-sources\protobuf\java
[ERROR] PROTOC FAILED: Error: Could not find or load main class C:\Program
Caused by: java.lang.ClassNotFoundException: C:\Program
--yamcs_out: protoc-gen-yamcs: Plugin failed with status code 1._

_[ERROR] C:\Program Files\yamcs\yamcs-master\yamcs-api\src\main\proto\yamcs\api\annotations.proto [0:0]: Error: Could not find or load main class C:\Program
Caused by: java.lang.ClassNotFoundException: C:\Program
--yamcs_out: protoc-gen-yamcs: Plugin failed with status code 1._

_[ERROR] C:\Program Files\yamcs\yamcs-master\yamcs-api\src\main\proto\yamcs\api\exception.proto [0:0]: Error: Could not find or load main class C:\Program
Caused by: java.lang.ClassNotFoundException: C:\Program
--yamcs_out: protoc-gen-yamcs: Plugin failed with status code 1._

Let me know what else can be of help that I can share...

**Reply by fqqb** [reply]
I see, it's the space in your path that is causing the issue.
I will fix it in a new version of yamcs-maven-plugin somewhere Today. Workaround would be to use a location without spaces.

**Reply by JohnAdam-123** [reply]
So, how can i solve it now. Or can you specify as how I can solve it or should I wait you fix it?

**Reply by fqqb** [reply]
This commit on master should fix your issue: https://github.com/yamcs/yamcs/commit/d4bba0b0a541b84ac1fb4f71fd596c53ab9da5f0

Or don't use a path with spaces in it (i.e. `Program Files`)

**Reply by JohnAdam-123** [reply]
Okay.
